### PR TITLE
Migrate json.md to null safety

### PR DESCRIPTION
**Pull request by miquelbeltran**
Here I have migrated the json.md to null safety. Almost all code examples have been moved to a project and imported using code-excerpts. There's one thing I am not sure if I like, which I am addressing below in the comments.

cc. @sfshaza2 @RedBrogdon

### CODE REVIEW

File: src/docs/development/data-and-backend/json.md [hunk 1]
```diff
@@ -278,18 +280,18 @@ from the previous samples.
...
 
-{% prettify dart %}
+<?code-excerpt "lib/serializable/user.dart"?>
+```dart
 import 'package:json_annotation/json_annotation.dart';
 
 /// This allows the `User` class to access private members in
 /// the generated file. The value for this is *.g.dart, where
 /// the star denotes the source file name.
-part '[[highlight]]user[[/highlight]].g.dart';
```
Comment (miquelbeltran) [inline src/docs/development/data-and-backend/json.md:287]: I am not sure how to add the `[[highlight]]` code in here, using a "replace" regex in the code-excerpt seems very complicated. Alternatively I could revert this change and leave this code sample without a code-excerpt. I am open to ideas!

Comment (sfshaza2) [reply src/docs/development/data-and-backend/json.md:287]: Yes, the highlight directive is only supported in `prettify` sections. In this case, just remove the highlight. I don't want you jumping hoops with regex. Just please make sure that the introductory text to this section doesn't refer to the "highlighted" code, since it won't be.

**Comment by sfshaza2** [comment]
Other than that, it LGTM. Can you take a look, @domesticmouse?

**Review by domesticmouse — APPROVED** [review]

File: src/docs/development/data-and-backend/json.md (hunk 1 above)
Comment (miquelbeltran) [reply src/docs/development/data-and-backend/json.md:287]: Got it! I will double check that

Comment (miquelbeltran) [reply src/docs/development/data-and-backend/json.md:287]: I didn't see any references to highlighted code, so it should be good to go :)
